manifests/install.pp

```diff
@@ -30,11 +30,26 @@
     extract_path => $aerospike::download_dir,
     creates      => $dest,
     cleanup      => $aerospike::remove_archive,
-  } ~>
-  exec { 'aerospike-install-server':
-    command     => "${dest}/asinstall",
-    cwd         => $dest,
-    refreshonly => true,
+    notify       => Exec['aerospike-install-server'],
```

Comment:
The notify is needed there. Even when you change user/password it's probably because the old ones didn't work. I've reverted it to the old syntax, see c17425b.